.github/workflows/run-tests.yml

```diff
@@ -136,15 +162,35 @@ jobs:
         run: |
           docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} down -v
 
+  set-up-functional-test:
+    name: Set up functional test
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - name: build functional test matrix
+        id: json
```

Comment:
what is this `id` for?

Reply:
at the time I had just copy-pasted it to play with, so I didn't know what it was for yet, but it's for identifying the step in the job outputs section. See https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#example-defining-outputs-for-a-job